File: spark/projects/simple-spark-project/src/main/scala/com/thedataincubator/simplespark/SimpleApp.scala
package com.thedataincubator.simplespark

/* SimpleApp.scala */
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._
import org.apache.spark.SparkConf

object SimpleApp {
  def main(args: Array[String]) {
    if (args.length != 1) {
        System.err.println("Requires HDFS input as argument")
        System.exit(1)
    }
    val logFile = args(0)
    val conf = new SparkConf().setAppName("Simple Application")
    val sc = new SparkContext(conf)
    val logData = sc.textFile(logFile, minPartitions=2).cache()
    val numAs = logData.filter(line => line.contains("a")).count()
    val numBs = logData.filter(line => line.contains("b")).count()
    println("#" * 80)
    println("Lines with a: %s, Lines with b: %s".format(numAs, numBs))
    println("#" * 80)
  }
}

File: spark/projects/simple-pyspark-project/SimpleApp.py
from pyspark import SparkContext, SparkConf
import sys
from datetime import datetime
from myutils import contains

def main(*args):
    if len(args) != 1:
        print "Requires a single HDFS path as argument"
        sys.exit(1)

    fn = args[0]
    conf = SparkConf()
    conf.setAppName("SimpleApp")
    sc = SparkContext(conf=conf)

    data = sc.textFile(fn, 2).cache()
    numAs = data.filter(contains("a")).count()
    numBs = data.filter(contains("b")).count()

    print "#" * 80
    print "Lines with a: {A}, Lines with b: {B}".format(A=numAs, B=numBs)
    print "#" * 80


if __name__ == '__main__':
    main(*sys.argv[1:])
